src/cluster.c

```diff
@@ -4597,44 +4639,60 @@ NULL
         /* Check that all the arguments are parseable and that all the
          * slots are not already busy. */
         for (j = 2; j < c->argc; j++) {
-            if ((slot = getSlotOrReply(c,c->argv[j])) == -1) {
+            if ((slot = getSlotOrReply(c,c->argv[j])) == C_ERR) {
```

Comment:
> Is it enough to only check the slots named in the argument list? Should getSlotOrReply be called for all the slots between the start and end of each range?

Line 4642 is for existing command: addslots and delslots, these 2 commands only accept individual slot.  Thus, every parameter in the argument list is an individual value, it is not a range.

In the Line 4672 and Line 4676, they check the start point and end point of the range.

Reply:
Oops, I meant to make this comment on line 4672.  At that point, shouldn't you check the slots inside the range also?

Reply:
> Oops, I meant to make this comment on line 4672. At that point, shouldn't you check the slots inside the range also?

I think we do not need to check the slots inside the range. Because in Line 4672 we check the start slot value, and in Line 4676 check the end slot value.  That means if start slot and end slot satisfy the condition (condition is the slot value is from 0 to 16383 ), the all slot values in the range will satisfy the condition automatically. 

This is why I do not check the slots inside the range, how do you think?

Reply:
Sorry for the confusion, I corrected my comment but I guess you didn't get notified of my correction soon enough.

I agree that you are doing enough checks.  `getSlotOrReply` is only checking that the slot number is reasonable, and only has to be done on actual arguments.  `getSlotStatus` is being used to check all the slots in the range.  So everything is good.